DependencyInjection/OrnicarGravatarExtension.php

```diff
@@ -18,6 +20,37 @@ public function load(array $configs, ContainerBuilder $container)
         $processor = new Processor();
         $config = $processor->process($configuration->getConfigTree(), $configs);
 
-        $container->getDefinition('gravatar.api')->addArgument($config );
+        $container->getDefinition('gravatar.api')->addArgument($config);
+
+        if (isset($config['cache']['adapter'])) {
+            $cache = $config['cache']['adapter'];
+            $container->getDefinition('gravatar.api')->addMethodCall(
+                'setCache',
+                array(
+                    new Reference(
+                        $cache,
+                        ContainerInterface::IGNORE_ON_INVALID_REFERENCE
```

Comment:
We shouldnt ignore it on invalid. If the user sets the cache we should fail hard if it doesnt exists.